### PR TITLE
fix(input): smart support of icons

### LINKED ISSUE #6348: mdInputContainer: Has padding-left using .md-icon-right with only md-icon on the right side
Version: 1.0.0
OS: Windows 10
Browser: Chrome 47

Right now  `md-input-container` has a padding left if it has the class `.md-icon-right`, even if the only icon present is on the right side.

    <md-input-container class="md-icon-right">
      <input type="text"></input>
      <md-icon md-font-set="material-icons">clear</md-icon>
    </md-input-container>

Example: http://codepen.io/anon/pen/wMMQde
you can see the top `md-input-container` is the entire width while the bottom one has a `padding-left`. The third one has the fix below.

This is easily fixed by adding a class like `md-icon-right-only` and using the following CSS

    md-input-container.md-icon-right.md-icon-right-only{
        padding-left: 0;
    }

    md-input-container.md-icon-right.md-icon-right-only > label{
         left: 0;
    }

This removes the `padding-left`from the `md-input-container` and repositions the `label` accordingly.

### LINKED ISSUE #6896: md-input-container render fix for only right icon
Hi, if I want to add an icon only on the right side of the input it doesn't render well:

render-off

Example structure:
```html
<md-input-container class="md-icon-right">
	<label>Price</label>
	<input type="text" ng-model="price"></input>
	<md-icon md-font-icon="mdi mdi-currency-eur"></md-icon>
</md-input-container>
```

Proposed CSS fix:
```css
.md-has-icon {
	&.md-icon-right {
		padding-left: 0;

		& > label {
			left: 0;
		}
	}
}
```

Renders correctly:

render-ok

Regards.

### PR DESCRIPTION
**Pull request by EladBezalel**
- Fixed single icon on the right
- Added smart classes applying, now the directive will automatically add md-icon-left/right according the icon placement (after/before the input)

fixes #6348. fixes #6896.

### CODE REVIEW

File: src/components/input/input.js [hunk 1]
```diff
@@ -62,7 +62,17 @@ function mdInputContainerDirective($mdTheming, $parse) {
...
-    if (element.find('md-icon').length) element.addClass('md-has-icon');
+
+    var iconElements = element.find('md-icon');
+    var icons = iconElements.length ? iconElements : element[0].getElementsByClassName('md-icon');
+    if (icons.length == 1) {
+      element.addClass(icons[0].nextElementSibling && icons[0].nextElementSibling.tagName === 'INPUT' ? 'md-icon-left' :
+                       icons[0].previousElementSibling && icons[0].previousElementSibling.tagName === 'INPUT' ? 'md-icon-right' :
+                         '');
+    }
+    else if (icons.length == 2){
```
Comment (DevVersion) [inline src/components/input/input.js]: I think the `){` is not by intention right?

Comment (EladBezalel) [inline src/components/input/input.js]: yea it should have space between :sweat_smile:

File: src/components/input/input.js [hunk 2]
```diff
@@ -62,7 +62,17 @@ function mdInputContainerDirective($mdTheming, $parse) {
 
   function postLink(scope, element, attr) {
     $mdTheming(element);
-    if (element.find('md-icon').length) element.addClass('md-has-icon');
+
+    var iconElements = element.find('md-icon');
+    var icons = iconElements.length ? iconElements : element[0].getElementsByClassName('md-icon');
+    if (icons.length == 1) {
+      element.addClass(icons[0].nextElementSibling && icons[0].nextElementSibling.tagName === 'INPUT' ? 'md-icon-left' :
```
Comment (ThomasBurleson) [inline src/components/input/input.js]: reduce clutter: use 

```
var previous = icons[0].previousElementSibling;
var next = icons[0].nextElementSibling;
```

File: src/components/input/input.js [hunk 3]
```diff
@@ -62,7 +62,17 @@ function mdInputContainerDirective($mdTheming, $parse) {
...
+                       icons[0].previousElementSibling && icons[0].previousElementSibling.tagName === 'INPUT' ? 'md-icon-right' :
+                         '');
+    }
+    else if (icons.length == 2) {
+      element.addClass('md-icon-left md-icon-right');
```
Comment (ThomasBurleson) [inline src/components/input/input.js]: Please add comment explaining why these classes are used.

**Comment by ThomasBurleson** [comment]
@EladBezalel - do we have a demo/sample to show this live?

**Comment by EladBezalel** [comment]
@ThomasBurleson in the icons section on the input demo

**Pull request closed without merging**